CsharpExtrasTest/Dictionary/MultiValueMapTest.cs

```diff
@@ -95,5 +95,51 @@ public void GivenMap_WhenTransform_ThenResultIsOfCorrectLengthAndContainsExpecte
             Assert.IsTrue(new HashSet<int>(new int[] {3}).SetEquals(transformedMap[2]));
             Assert.IsTrue(new HashSet<int>(new int[] {5}).SetEquals(transformedMap[3]));
         }
+
+        [Test]
+        [Category("Unit")]
+        public void Given_TwoMaps_When_TheSameDataIsAddedInDifferentOrders_Then_MapHashCodesMatch()
+        {
+            //Assemble
+            IMultiValueMap<int, string> map1 = new MultiValueMapImpl<int, string>();
+            IMultiValueMap<int, string> map2 = new MultiValueMapImpl<int, string>();
+
+            //Act
+            map1.Add(1, "a");
+            map1.Add(1, "b");
+            map1.Add(1, "c");
+            map1.Add(2, "d");
+            map1.Add(2, "e");
+
+            map2.Add(2, "d");

```

Comment:
The order of d and e is still the same here as above, although interleaved with other statements. I'd swap them.